.github/actions/setup-gstreamer/action.yml

```
@@ -0,0 +1,177 @@
+            echo "Checking for runtime installer under s3://${bucket}/${prefix}/"
+            runtime_msi=$(aws s3api list-objects-v2 --bucket "${bucket}" --prefix "${prefix}/" --query 'Contents[].Key' | jq -r 'if type!="array" then [] else . end | sort_by(length) | map(select(endswith("'"${RUNTIME_MSI_FILENAME}"'"))) | .[0]')
+            if [ "$runtime_msi" != "null" ]; then
+              echo "runtime-url=$(aws s3 presign s3://${bucket}/${runtime_msi})" | tee -a $GITHUB_OUTPUT
+            fi
+          fi
+
+          if [ "$dev_msi" = "null" ]; then
+            echo "Checking for development installer under s3://${bucket}/${prefix}/"
+            dev_msi=$(aws s3api list-objects-v2 --bucket "${bucket}" --prefix "${prefix}/" --query 'Contents[].Key' | jq -r 'if type!="array" then [] else . end | sort_by(length) | map(select(endswith("'"${DEV_MSI_FILENAME}"'"))) | .[0]')
+            if [ "$dev_msi" != "null" ]; then
+              echo "dev-url=$(aws s3 presign s3://${bucket}/${dev_msi})" | tee -a $GITHUB_OUTPUT
+            fi
+          fi
+
+          if [ "$runtime_msi" != "null" ] && [ "$dev_msi" != "null" ]; then
+            break
+          fi
+        done
+
+      # If either installer was not found, run a build.
+    - id: build
+      if: ${{ !steps.check-installer-exists.outputs.runtime-url || !steps.check-installer-exists.outputs.dev-url }}
+      name: Build GStreamer with Cerbero on Windows
+      uses: ./.github/actions/build-gstreamer
+      with:
+        config: ${{ inputs.config }}
+        cerbero-args: ${{ inputs.cerbero-args }}
+        cerbero-package-args: ${{ inputs.cerbero-package-args }}
+        cerbero-package-origin: ${{ steps.version-info.outputs.gstreamer-version }}-${{ steps.version-info.outputs.cerbero-short-sha }}
+        bootstrap-system: ${{ inputs.bootstrap-system }}
+        gst-plugins-rs-repo: ${{ inputs.gst-plugins-rs-repo }}
+        gst-plugins-rs-ref: ${{ inputs.gst-plugins-rs-ref }}
+        runtime-msi-filename: ${{ steps.version-info.outputs.runtime-msi-filename }}
+        dev-msi-filename: ${{ steps.version-info.outputs.dev-msi-filename }}
+        force: ${{ inputs.force }}
+        cleanup: ${{ inputs.cleanup }}
+        no-cache: ${{ inputs.no-cache }}
+
+    # Install either the local, pre-existing, or newly-built gstreamer MSIs
+    - id: install
+      name: Install GStreamer on Windows
+      uses: ./.github/actions/install-gstreamer
+      with:
+        runtime-installer-url: ${{ steps.check-installer-exists.outputs.runtime-url || steps.build.outputs.runtime-installer-url }}
+        runtime-installer-path: ${{ steps.build.outputs.runtime-installer-path }}
+        development-installer-url: ${{ steps.check-installer-exists.outputs.dev-url || steps.build.outputs.devel-installer-url }}
+        development-installer-path: ${{ steps.build.outputs.devel-installer-path }}
+
+    - name: Cleanup Cerbero
+      shell: msys2 -eo pipefail {0}
```

Should be gated by the `cleanup` input:
```suggestion
    - name: Cleanup Cerbero
      if: ${{ always() && inputs.cleanup == 'true' && steps.get-cerbero.outcome != 'skipped' }}
      shell: msys2 -eo pipefail {0}
```